src/settingsui.ts

```diff
@@ -533,19 +595,144 @@ export function renderSettings() {
 				};
 				if (userscript.meta) { // render custom metadata if provided
 					const thisMeta = userscript.meta;
+					// Define low-scope variables because I can't be arsed to copy + paste the same ternary operator
+					var scriptAuthor = ('author' in thisMeta && thisMeta.author) ? `${thisMeta.author}` : false
+					var scriptName = ('name' in thisMeta && thisMeta.name) ? thisMeta.name : userscript.name
 					Object.assign(obj, {
-						title: 'name' in thisMeta && thisMeta.name ? thisMeta.name : userscript.name,
+						title: scriptName,
 						desc: `${'desc' in thisMeta && thisMeta.desc ? thisMeta.desc.slice(0, 60) : ''}
-						${'author' in thisMeta && thisMeta.author ? `&#8226; ${thisMeta.author}` : ''}
+						${scriptAuthor ? `&#8226; ${scriptAuthor}` : ''}
 						${'version' in thisMeta && thisMeta.version ? `&#8226; v${thisMeta.version}` : ''}
-						${'src' in thisMeta && thisMeta.src ? ` &#8226; <a target="_blank" href="${thisMeta.src}">source</a>` : ''}`
+						${'src' in thisMeta && thisMeta.src ? ` &#8226; <a target="_blank" href="${thisMeta.src}">source</a>` : ''}
+						${'settings' in userscript && Object.keys(userscript.settings).length > 0 && su.userscriptTracker[userscript.name] ? `&#8226; Uses ${Object.keys(userscript.settings).length} Custom Settings` : ``}`
 					});
+					// Read and render script-defined settings
+					if ('settings' in userscript && Object.keys(userscript.settings).length > 0 && su.userscriptTracker[userscript.name]) {
+						// Create separate fragment so that script settings go at the bottom of the client page.
+						const brokenSettings: string[] = []
+						var fragForUserscript: DocumentFragment = new DocumentFragment()
+						var userscriptCategoryID: string = `${scriptName}by${scriptAuthor}`.replaceAll(' ', '').toLowerCase().replaceAll(/[^a-z0-9]/g, '')
+						fragForUserscript.appendChild(skeleton.catHedElem(` ${scriptName} <span class='settings-Userscript-Author'>by ${scriptAuthor}</span>`));
+						fragForUserscript.appendChild(skeleton.catBodElem(userscriptCategoryID, ``));
+						try {
+							var userScriptDefinedOptions: { [key: string]: UserscriptRenderReadySetting } = {... userscript.settings}
```

Comment:
should be the same thing, test this afterwards tho
```suggestion
							var userScriptDefinedOptions: Record<string, UserscriptRenderReadySetting> = {... userscript.settings}
```